src/vm.rs

```diff
@@ -396,10 +410,10 @@ pub type InitializingVm<'a, T> = VmRef<'a, T, VmStateInitializing>;
 
 impl<'a, T: GuestStagePagingMode> InitializingVm<'a, T> {
     /// Adds a vCPU to this VM.
-    fn add_vcpu(&self, vcpu_id: u64, shared_area: VmCpuSharedArea) -> EcallResult<()> {
+    fn add_vcpu(&self, vcpu_id: u64, vcpu_box: PageBox<Mutex<VmCpu>>) -> EcallResult<()> {
```

Comment:
Need some rust lessons here :). What does PageBox represent ?

Reply:
ah, not a general rust thing. PageBox is a salus-ism and I'm to blame. It's a smart pointer like the standard rust `Box` but backed by a page or pages instead of a general purpose allocation.

Reply:
Thanks. I am reading the comments now.
https://github.com/rivosinc/salus/blob/main/page-tracking/src/collections/page_box.rs